charts/taraxa-node/templates/transaction-generation-service.yaml

```diff
@@ -0,0 +1,73 @@
+{{ if .Values.transactionGeneration.enabled }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Release.Name }}-transaction-generation
+  labels:
+    app: transaction-generation
+    app.kubernetes.io/name: transaction-generation
+    helm.sh/chart: {{ include "taraxa-node.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  replicas: 1
+  serviceName: {{ .Release.Name }}-transaction-generation
+  # to launch or terminate all Pods in parallel.
+  # https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#parallel-pod-management
+  podManagementPolicy: Parallel
+  selector:
+    matchLabels:
+      app: transaction-generation
+      app.kubernetes.io/name: {{ .Release.Name }}-transaction-generation
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      name: transaction-generation
+      labels:
+        app: transaction-generation
+        app.kubernetes.io/name: {{ .Release.Name }}-transaction-generation
+        app.kubernetes.io/instance: {{ .Release.Name }}
+      annotations:
+        kubernetes.io/change-cause: "Configuration through configmaps."
+    spec:
+      containers:
+        - name: transaction-generation
+          image: "python:3.8"
+          imagePullPolicy: IfNotPresent
+          env:
+          - name: PRIVATE_KEY
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Release.Name }}
+                key: TRANSACTION_GENERATION_PRIVATE_KEY
```

Comment:
can't we use the same name var name `PRIVATE_KEY` and key in secret ?

Reply:
The secret is for the whole taraxa-node deployment. There are a lot of private keys there.